pandas/core/arrays/string_.py

```diff
@@ -192,6 +181,12 @@ def __arrow_array__(self, type=None):
             type = pa.string()
         return pa.array(self._ndarray, type=type, from_pandas=True)
 
+    def _values_for_factorize(self):
+        arr = self._ndarray.copy()
+        mask = self.isna()
+        arr[mask] = -1
+        return arr, -1
```

Comment:
You can't specify pd.NA here as the indicator? (since that is already in the values) 
Or does the algo code does not like that?

Reply:
Algos didn't like it. Somewhere in there we do a `value == na_value`, which raises.

We may need / want to rewrite things to be masked based.